src/main/java/com/arpnetworking/metrics/mad/model/DefaultRecord.java

```diff
@@ -181,16 +193,28 @@ public Builder setAnnotations(final ImmutableMap<String, String> value) {
             return this;
         }
 
+        /**
+         * The dimension mappings <code>ImmutableMap</code>. Optional. Default is an empty
+         * <code>ImmutableMap</code>. Cannot be null.
+         *
+         * @param value The dimension mappings <code>ImmutableMap</code>
+         * @return This instance of <code>Builder</code>.
+         */
+        public Builder setDimensions(final ImmutableMap<String, String> value) {
+            _dimensions = value;
+            return this;
+        }
+
         // Called by OVal reflectively
         @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD")
-        private boolean validateAnnotations(final ImmutableMap<String, String> annotations) {
-            if (Strings.isNullOrEmpty(annotations.get(Key.HOST_DIMENSION_KEY))) {
+        private boolean validateDimensions(final ImmutableMap<String, String> dimensions) {
+            if (Strings.isNullOrEmpty(dimensions.get(Key.HOST_DIMENSION_KEY))) {
```

Comment:
So the goal should be to remove these. I *believe* there is no code in MAD relying on them. However, we should do this in a separate PR. Please file an issue for MAD to this effect.

Additionally, the equivalent code in CAgg needs to be generalized. @matthayter has started this work but his most recent PR still references these fields explicitly. Matt, can you file a corresponding issue for CAgg to use a configurable set of dimensions to aggregate over and drop these specific ones from the model?

Finally, please cross-link the two issues.

@BrandonArp any thoughts to add?